### PR TITLE
Boss #1 Bomb Boss

### PR DESCRIPTION
**Pull request by kblack89**
Boss Bomb attack being merged into Boss scaling branch

### CODE REVIEW

**Comment by kblack89** [comment]
This is ready for review.

File: Assets/Scripts/EnemyScripts/EnemyExplosion.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        yield return new WaitForSeconds(0.25f);
+        Destroy(gameObject);
+    }
+
+    void OnTriggerEnter2D(Collider2D other)
+    {
+        if (other.gameObject.CompareTag("Player"))
+        {
+            other.gameObject.GetComponent<PlayerStats>().ReceiveDamage(10);
+            Destroy(gameObject);
```
Comment (AlexReidNZ) [inline Assets/Scripts/EnemyScripts/EnemyExplosion.cs:46]: I would maybe not destroy the explosion here, as if you get hit by the explosion when it explodes, it gets destroyed instantly and you can't see the explosion that damaged you. The explosions are short enough that we probably won't need to worry about the player walking in and out of it and taking damage more than once, but if you wanted to fix that you could disable the collider when it damages the player (see melee enemy attack).

**Review by AlexReidNZ — CHANGES_REQUESTED** [review]
The new boss behaviour is really cool, the bombs work as expected and are fun to fight. A couple minor things I noticed:
* The EnemyBoss1 prefab is not assigned to the spawner, there is an empty slot
* The EnemyBoss1 prefab has a doubled fire rate compared to the other bomb boss, it shoots really fast.

**Comment by kblack89** [comment]
I have made the requested changes and tested out. This is ready for re-review.

**Review by AlexReidNZ — APPROVED** [review]
After removing the redundant code, this is much more suitable for merging.

**Review by PBWiseman — APPROVED** [review]
I love the boss design! After the boss mechanics change we discussed in person, I am happy for this to be merged

**Review by Loerad — APPROVED** [review]
We will remove the debug.logs from the script in staging

**Pull request merged**